### PR TITLE
[DynamicType] Add compilation time and memory usage benchmark

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Just trying to pull some numbers for compilation speed and memory usage. See rendered README at for result:

https://github.com/NVIDIA/Fuser/tree/compilation-speed/lib/dynamic_type#compilation-time

### CODE REVIEW

File: lib/dynamic_type/README.md [hunk 1]
```diff
@@ -262,3 +262,20 @@ dynamic structs should generally not be implemented as a dictionary. Please refe
 semi-dynamic struct.
 
 # Compilation time
+
+The compilation time and memory usage are exponential with respect to the number of dynamic types.
```
Comment (zasdfgbnm) [inline lib/dynamic_type/README.md:266]: Maybe cubic, not exponential

Comment (zasdfgbnm) [reply lib/dynamic_type/README.md:266]: Hmm, looks like exponential has the best fit compared to quadratic vs cubic:
image
I think we can conclude it's exponentional.

**Review by naoyam — APPROVED** [review]
So, clang is much faster than gcc? Which versions are they?

**Comment by zasdfgbnm** [comment]
> So, clang is much faster than gcc? Which versions are they?

Yes, clang is much faster in terms of time. It also uses less memory. Versions are g++ 13.2.1, clang++ 15.0.7

**Pull request merged**